app/sample/ext_frame/rx_st20_pipeline_dyn_ext_frame_sample.c

```diff
@@ -49,6 +49,14 @@ static int rx_st20p_query_ext_frame(void* priv, struct st_ext_frame* ext_frame,
   ext_frame->iova[0] = s->ext_frames[i].buf_iova;
   ext_frame->size = s->ext_frames[i].buf_len;
 
+  uint8_t* addr = ext_frame->addr[0];
+  uint8_t planes = st_frame_fmt_planes(meta->fmt);
+  for (int plane = 0; plane < planes; plane++) {
+    ext_frame->linesize[plane] = st_frame_least_linesize(meta->fmt, meta->width, plane);
+    ext_frame->addr[plane] = addr;
+    addr += ext_frame->linesize[plane] * meta->height;
```

Comment:
Different planes might have different heights; for example, YUV420. You should take this into consideration, whereas here you use meta->height, which is constant across all planes.